src/test/java/io/kontur/disasterninja/dto/layer/LayerSummaryDtoTest.java

```diff
@@ -0,0 +1,185 @@
+package io.kontur.disasterninja.dto.layer;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.kontur.disasterninja.domain.Layer;
+import io.kontur.disasterninja.util.JsonUtil;
+import org.junit.jupiter.api.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class LayerSummaryDtoTest {
+
+    @Test
+    public void testMapBoxStyle_userSettingsPositive() {
+        Layer layer = createLayer();
+        layer.setMapboxStyles(JsonUtil.readJson("{\"url\":\"https://d.n/style_{lang}.json\",\"lang\":[\"kz\",\"ar\"]}", ObjectNode.class));
+
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.addHeader("accept-language", "en-BY,en;q=0.9,ru-BY;q=0.8,ru;q=0.7,en-GB;q=0.6,en-US;q=0.5");
+        request.addHeader("user-language", "ar");
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(request));
+
+        LayerSummaryDto dto = LayerSummaryDto.fromLayer(layer);
+
+        assertEquals("https://d.n/style_ar.json", dto.getMapboxStyle());
+    }
+
+    @Test
+    public void testMapBoxStyle_noUserSettings() {
+        Layer layer = createLayer();
+        layer.setMapboxStyles(JsonUtil.readJson("{\"url\":\"https://d.n/style_{lang}.json\",\"lang\":[\"en\",\"ar\"]}", ObjectNode.class));
+
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.addHeader("accept-language", "ar,en;q=0.9,ru-BY;q=0.8,ru;q=0.7,en-GB;q=0.6,en-US;q=0.5");
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(request));
+
+        LayerSummaryDto dto = LayerSummaryDto.fromLayer(layer);
+
+        assertEquals("https://d.n/style_ar.json", dto.getMapboxStyle());
+    }
+
+    @Test
+    public void testMapBoxStyle_AcceptLanguageFirstWin() {
+        Layer layer = createLayer();
+        layer.setMapboxStyles(JsonUtil.readJson("{\"url\":\"https://d.n/style_{lang}.json\",\"lang\":[\"en\",\"ar\"]}", ObjectNode.class));
+
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.addHeader("accept-language", "ar,en;q=0.9,ru-BY;q=0.8,ru;q=0.7,en-GB;q=0.6,en-US;q=0.5");
```

Comment:
wait, does this mean that basemap will be translated if frontend language is not selected? seems wrong. we don't control accept-language and thus if front end forgets to set user-language then you will get heisenbugs that depend on user's computer's language.

I'd rather let us ERROR in the logs if no user-language provided. The logic of choosing app locale based on browser locale may need to go to frontend instead, once and overridable.

Reply:
> wait, does this mean that basemap will be translated if frontend language is not selected?

Wasn't it initial plan to support accept-language with the list of languages?

>  if front end forgets to set user-language 

You know what? We don't set any language by default for non logged users and for users who didn't set language explicitly in app settings. So basemap will be translated according to accept-language for these cases. You are right. Seems like not the best solution.